hal/src/dmac/transfer.rs

```diff
@@ -548,6 +548,18 @@ where
         self.chan.as_mut().software_trigger();
     }
 
+    /// Unsafely and mutably borrow the source buffer
```

Comment:
Do these need  safety comments per the clippy PR?

Reply:
Indeed, fixed!